src/applications/vaos/reducers/newAppointment.js

```diff
@@ -659,7 +669,10 @@ export default function formReducer(state = initialState, action) {
 
       return {
         ...state,
-        data,
+        data: {
```

Comment:
Instead of the standalone clear data action, I've added this to clear the calendarData if the user goes back to the clinic selection page.